package/samplers/auto_sampler/_sampler.py

```diff
@@ -0,0 +1,232 @@
+
+    def reseed_rng(self) -> None:
+        self._rng.rng.seed()
+        self._sampler.reseed_rng()
+
+    def _include_conditional_param(self, study: Study) -> bool:
+        trials = study._get_trials(
+            deepcopy=False, states=(TrialState.COMPLETE, TrialState.PRUNED), use_cache=True
+        )
+        if len(trials) == 0:
+            return False
+
+        param_key = set(trials[0].params)
+        return any(param_key != set(t.params) for t in trials)
+
+    def _determine_multi_objective_sampler(
+        self, study: Study, trial: FrozenTrial, search_space: dict[str, BaseDistribution]
+    ) -> BaseSampler:
+        # TODO(nabenabe): Add more efficient heuristic for MO.
+        if isinstance(self._sampler, NSGAIISampler):
+            return self._sampler
+
+        seed = self._rng.rng.randint(_MAXINT32)
+        return NSGAIISampler(constraints_func=self._constraints_func, seed=seed)
+
+    def _determine_single_objective_sampler(
+        self, study: Study, trial: FrozenTrial, search_space: dict[str, BaseDistribution]
+    ) -> BaseSampler:
+        if isinstance(self._sampler, TPESampler):
+            return self._sampler
+
+        seed = self._rng.rng.randint(_MAXINT32)
+        if (
+            self._constraints_func is not None
+            or any(isinstance(d, CategoricalDistribution) for d in search_space.values())
+            or self._include_conditional_param(study)
+        ):
+            # Use ``TPESampler`` if search space includes conditional or categorical parameters.
+            # TBD: group=True?
+            return TPESampler(
+                seed=seed,
+                multivariate=True,
+                warn_independent_sampling=False,
+                constraints_func=self._constraints_func,
+                constant_liar=True,
+            )
+
+        complete_trials = study._get_trials(
+            deepcopy=False, states=(TrialState.COMPLETE,), use_cache=True
+        )
+        if len(complete_trials) < self._N_COMPLETE_TRIALS_FOR_CMAES:
+            # Use ``GPSampler`` if search space is numerical and
+            # len(complete_trials) < _N_COMPLETE_TRIALS_FOR_CMAES.
+            if not isinstance(self._sampler, GPSampler):
+                return GPSampler(seed=seed)
+        elif not isinstance(self._sampler, CmaEsSampler):
+            # Use ``CmaEsSampler`` if search space is numerical and
+            # len(complete_trials) > _N_COMPLETE_TRIALS_FOR_CMAES.
+            # Warm start CMA-ES with the first _N_COMPLETE_TRIALS_FOR_CMAES complete trials.
+            complete_trials.sort(key=lambda trial: trial.datetime_complete)
+            warm_start_trials = complete_trials[: self._N_COMPLETE_TRIALS_FOR_CMAES]
+            # NOTE(nabenabe): ``CmaEsSampler`` internally falls back to ``RandomSampler`` for
+            # 1D problems.
+            return CmaEsSampler(
+                seed=seed, source_trials=warm_start_trials, warn_independent_sampling=False
```

Comment:
I think that `warn_independent_sampling` should be `True`. For `AutoSampler`, it only works for 1-D objectives, so the warning is intended in this case.